sources/minishell.c

```diff
@@ -56,5 +56,6 @@ int	main(int argc, char **argv, char **envp)
 	shell = NULL;
 	shell = shell_pre_init(shell, envp, argv);
 	shell_run(shell);
+	// ? how to clean up pre-allocated memory from shell_pre_init? f.e. free after exit?
```

Comment:
Exit should return control to main, will be able to free. Rather add this comment to the ticket instead of in the code.